src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt

```diff
@@ -0,0 +1,173 @@
+package net.maizegenetics.phgv2.agc
+
+import biokotlin.util.bufferedReader
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.validate
+import com.github.ajalt.clikt.parameters.types.int
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import org.apache.logging.log4j.LogManager
+import java.io.File
+
+/**
+ * This class takes a keyfile containing a list of fasta files and their sample names.
+ * The fasta files should be the full path name to the fasta file.
+ * The sample name is the name that identifies the genome to be loaded.  This name should
+ * be consistent with the sample names used in the GVCF files for these genomes.
+ *
+ * Fasta files may be compressed or uncompressed.  If compressed, the extension should be .gz.
+ * The output directory is the directory where the updated fasta files will be written.
+ *
+ * THis class will take the fasta files, read them into memory, update the idlines with "sampleName=<sampleName>"
+ * The updated fasta files are written to the output directory with <sampleName>.fa as the file name.  If the original
+ * file was compressed, gzip compression will be performed on the new file.
+ *
+ * "threads" is an optional parameter.  This value determines how many fasta files will be processed concurrently.
+ * When determining the number of threads to use, keep in mind the memory capacity of the machine,
+ * and the size of the fasta files.  Each fasta is read into memory, idlines updated, then re-written
+ * to a new file of the same name in the user specified output directory.
+ *
+ */
+class AnnotateFastas : CliktCommand() {
```

Comment:
Add a help message here so it shows up from command line.

Reply:
added